### PR TITLE
Aruco board creation feature from 695

### PR DESCRIPTION
**Pull request by ShT3ch**
Implementation of requested feature #695

### CODE REVIEW

**Comment by paroj** [comment]
* check intendation
* comment the new API
* squash all commits into one

**Comment by ShT3ch** [comment]
@paroj done.. or not?

File: modules/aruco/src/aruco.cpp [hunk 1]
```diff
@@ -1330,6 +1329,33 @@ void GridBoard::draw(Size outSize, OutputArray _img, int marginSize, int borderB
 
 
 /**
+*/
+Ptr<Board> Board::create(InputArrayOfArrays objPoints, Ptr<Dictionary> &dictionary, InputArray ids) {
+
+    Mat ids_mat = ids.getMat();
+    std::vector<int> ids_vector;
+    for (int i = 0; i < ids_mat.rows; i++) {
+        ids_vector.push_back(ids_mat.at<int>(i));
```
Comment (paroj) [inline modules/aruco/src/aruco.cpp]: I think you can just do `ids.copyTo(ids_vector)` here. Or even better: `ids.copyTo(res->ids)`

File: modules/aruco/include/opencv2/aruco.hpp [hunk 2]
```diff
@@ -226,6 +226,16 @@ CV_EXPORTS_W void estimatePoseSingleMarkers(InputArrayOfArrays corners, float ma
...
+    /**
+    * @brief Provide way to create Board by passing nessesary data. Specially needed in Python.
+    *
+    * @param objPoints array of object points of all the marker corners in the board
+    * @param dictionary the dictionary of markers employed for this board
+    * @param ids vector of the identifiers of the markers in the board
+    *
+    * Provide way to create Board by passing nessesary data. Specially needed in Python.
+    */
+	CV_WRAP static Ptr<Board> create(InputArrayOfArrays objPoints, Ptr<Dictionary> &dictionary, InputArray ids);
```
Comment (paroj) [inline modules/aruco/include/opencv2/aruco.hpp]: bad intendation

File: modules/aruco/src/aruco.cpp [hunk 3]
```diff
@@ -1330,6 +1329,33 @@ void GridBoard::draw(Size outSize, OutputArray _img, int marginSize, int borderB
...
+    }
+
+    std::vector< std::vector< Point3f > > obj_points_vector;
+    for (unsigned int i = 0; i < objPoints.total(); i++) {
+        std::vector<Point3f> corners;
+        Mat corners_mat = objPoints.getMat(i);
+        for (int j = 0; j < 4; j++) {
+            corners.push_back(corners_mat.at<Point3f>(j));
```
Comment (paroj) [inline modules/aruco/src/aruco.cpp]: You could safe the temporary `corners_mat` by using `objPoints.at<Point3f>(i, j)`.

**Comment by sovrasov** [comment]
Replaced by #771

**Pull request closed without merging**